roachprod,cli: metric name consolidation in tsdump upload

aa-joshi opened the pull request:
Previously, tsdump was uploading time series data with metrics name defined in
registry. This was inadequate because same metric has different name in the
Datadog integration based on offering. We are working on consolidating
observability stack across our offerings. To address this, this patch updates
the tsdump upload functionality generate metrics renaming during execution. We
have updated gen-cockroachdb-metrics tool to generate yaml files. The
generation will support comprehensive and datadog only mode. In the
comprehensive mode, we are generating crdb metrics renaming mapping. The
datadog only mode would only generate the datadog integration metric renaming
mapping file.  We had to created datadog only mode to avoid circular dependency
on package cli.

Part of: CRDB-56324
Epic: None
Release note: None

Abhinav1299 commented on build/tools/gen-cockroachdb-metrics/main.go:
```diff
@@ -9,14 +9,14 @@
+// 1. Live metrics from test server
+//   - The primary source containing all metrics emitted by the current CRDB version
+//   - Collected by starting a test server and querying the ChartCatalog RPC endpoint
+//   - Represents the complete set of metrics available in the codebase
```
Can we highlight here that, the same method of collecting live metrics from spinning up the test server is used to document all the metrics in `docs/generated/metrics/metrics.yaml`? 
Reason being, the feature of automated metric propagation was announced with source of truth as metrics.yaml. Now removing its presence from this tool will be little confusing.

aa-joshi replied:
This comment is irrelevant now as we are still relying on metrics.yaml
<!-- Reviewable comment -OezwCcVENKqlumX4eCf:b-1y2uu7 -->
<!-- Sent from Reviewable.io -->

Abhinav1299 commented on build/tools/gen-cockroachdb-metrics/main.go:
```diff
@@ -122,25 +132,25 @@ const goTemplate = `// Copyright 2024 The Cockroach Authors.
+//      - Collected by starting a test server and querying metrics via RPC
+//      - Processed through Datadog mapping rules
+//   2. runtime_conditional_metrics from cockroachdb_metrics_base.yaml
+//      - Conditionally emitted metrics not available from test server
```
same as above comment to reference metrics.yaml

Abhinav1299 reviewed (commented):
Can we also confirm once the expected flow of the `gen-cockroachdb-metric` tool is not hampered by creating and registering a new test metric?

aa-joshi committed:
roachprod,cli: metric name consolidation in tsdump upload

Previously, tsdump was uploading time series data with metrics name defined in
registry. This was inadequate because same metric has different name in the
Datadog integration based on offering. We are working on consolidating
observability stack across our offerings. To address this, this patch updates
the tsdump upload functionality generate metrics renaming during execution. We
have updated gen-cockroachdb-metrics tool to generate yaml files. The
generation will support comprehensive and datadog only mode. In the
comprehensive mode, we are generating crdb metrics renaming mapping. The
datadog only mode would only generate the datadog integration metric renaming
mapping file.  We had to created datadog only mode to avoid circular dependency
on package cli.

Part of: CRDB-56324
Epic: None
Release note: None

aa-joshi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/158246)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @Abhinav1299, @arjunmahishi, @nameisbhaskar, and @shailendra-patel)



<!-- Sent from Reviewable.io -->

aa-joshi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/158246)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @Abhinav1299, @arjunmahishi, @nameisbhaskar, and @shailendra-patel)



<!-- Sent from Reviewable.io -->

Abhinav1299 reviewed (approved):
Looks like there is a merge conflict on the auto-generated mapping file, can you have a look at that.
Otherwise LGTM
